labs/src/main/scala/org/scalalabs/basic/lab02/ListManipulationExercise01.scala

```diff
@@ -99,7 +99,12 @@ object ListManipulationExercise01 {
    * Implement it whatever way suites you best. Hint: it can be done in a neat way using recursion. 
    */
   def tails[T](l: List[T]): List[List[T]] = {
-    error("fix me")
+    if(l.isEmpty){
+      List(l)
+    }
+    else{
+      List(l) ++ tails(l.tail)
+    }
```

Comment:
I did a lot of control statements for the first few parts of the lab as well (I think for this problem too, don't quite remember). I'm not sure if this is Scala-y, as it seems the provided solutions used mostly pattern matching, but it works so...